### DISCUSSION TITLE
Elastic Agent enrollment fail Service Unavailable

### DISCUSSION THREAD

**Discussion by Gregg3** [Security-Onion-Solutions/securityonion#11262] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

28

### RAM

384 GB

### Storage for /

293 GB

### Storage for /nsm

1.2 TB

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

We installed a ManagerSearch node from the ISO, but I'm having trouble getting my first windows client to install the Elastic agent. Here is the log of the failed agent installation:
```
  SO-Elastic-Agent_Installer.log 

timestamp=2023-09-06T11:25:11.3567582-04:00 level=info message="Version Information" ElasticAgentVersion=8.7.0 WrapperVersion=2.4.2
timestamp=2023-09-06T11:25:11.3578614-04:00 level=info message="Runtime Data" EnrollmentToken="RDdFUFRJb0JFakRNQU8xUXB4ekQ6WEUyMnhCSjdTUlNQNUlsQkdfVXlXZw==" FleetURL/s=https://so-manager.berriencounty.org:8220,https://so-manager:8220
timestamp=2023-09-06T11:25:11.3578614-04:00 level=info message="Installation Progress" Status="Starting Installation Precheck"
timestamp=2023-09-06T11:25:11.3820457-04:00 level=info message="Installation Progress" FleetHostConnectivityCheck=Success FleetHostURL=https://so-manager.berriencounty.org:8220
timestamp=2023-09-06T11:25:11.3837406-04:00 level=info message="Installation Progress" Status="Fleet Host is accessible - Continuing installation"
timestamp=2023-09-06T11:25:11.3837756-04:00 level=info message="Installation Progress" Status="Installation Precheck Complete"
timestamp=2023-09-06T11:25:11.3837756-04:00 level=info message="Installation Progress" Status="Extracting Elastic Agent files"
timestamp=2023-09-06T11:25:19.8917456-04:00 level=info message="Installation Progress" Status="Executing Elastic Agent installer"
timestamp=2023-09-06T11:25:19.8917456-04:00 level=info message="Installation Progress" Status="Executing the following: ./so-elastic-agent_source/elastic-agent/elastic-agent install --url=https://so-manager.berriencounty.org:8220 --enrollment-token=RDdFUFRJb0JFakRNQU8xUXB4ekQ6WEUyMnhCSjdTUlNQNUlsQkdfVXlXZw== --certificate-authorities=C:\\Program Files\\Elastic\\SO\\soca.crt -n"
timestamp=2023-09-06T11:26:18.5965056-04:00 level=error message="Installation Progress" Context="Installing in non-interactive mode.{\"log.level\":\"info\",\"@timestamp\":\"2023-09-06T11:26:11.895-0400\",\"log.origin\":{\"file.name\":\"cmd/enroll_cmd.go\",\"file.line\":478},\"message\":\"Starting enrollment to URL: https://so-manager.berriencounty.org:8220/\",\"ecs.version\":\"1.6.0\"}\nError: fail to enroll: fail to execute request to fleet-server: Service Unavailable\nFor help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.8/fleet-troubleshooting.html\nError: enroll command failed with exit code: 1\nFor help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/8.8/fleet-troubleshooting.html\n" ErrorDetails="exit status 1"
```
I've also tried using the "-fleet" option with ip address or hostname.
From other posts in this forum, it seems that my problem might be the related to the fact that I installed the ManagerSearch node using a proxy and specifying communications via FQDN. Is that scenario not working correctly? Do I need to reinstall?
```
   sudo salt-call pillar.get global 

local:
    ----------
    airgap:
        False
    endgamehost:
    fleet_grid_enrollment_token_general:
        <snip>
    fleet_grid_enrollment_token_heavy:
        <snip>
    ids:
        Suricata
    imagerepo:
        security-onion-solutions
    influxdb_host:
        so-manager
    managerip:
        10.10.8.2
    mdengine:
        ZEEK
    pipeline:
        redis
    registry_host:
        so-manager
    repo_host:
        so-manager
    soversion:
        2.4.10
    url_base:
        so-manager.<snip>
```

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by plaotec** [comment]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/f447b30a-7595-432b-9e3f-68ba94c36ce4)

we have problem also getting the agent to work

**Comment by plaotec** [comment]
![Screenshot 2023-09-06 145248](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/d9bdeba2-41dd-4afe-8e97-69cd16211ca2)

**Comment by plaotec** [comment]
![Screenshot 2023-09-06 145321](https://github.com/Security-Onion-Solutions/securityonion/assets/63425694/e3e51ddb-8bac-4871-b86f-d635426cb4ae)
[Uploading docs-securityonion-net-en-2.4.pdf…]()

**Comment by defensivedepth** [comment]
@plaotec Please open up a separate discussion

**Comment by OzzyKampha** [comment]
Have you allowed that host to connect in the firewall?

**Reply by plaotec** [reply]
Yes GUI firewall config above .. thanks

**Reply by Gregg3** [reply]
Me too. 
Firewall for elastic_agent_endpoint is configured as 172.16.0.0/15 - and test windows agent installation is on a machine in that range

**Comment by defensivedepth** [comment, marked as answer]
@Gregg3 So you have added a FQDN? Does this sound like what you are running into?

https://github.com/Security-Onion-Solutions/securityonion/discussions/11020

**Reply by Gregg3** [reply]
I've got to admit that #11020 wasn't entirely clear to me regarding what I should do to resolve the problem.

I went into the gui - Administration / Configuration / manager / no_proxy .. added fqdn  and applied
error log still shows "Service Unavailable"

I went into the gui - Administration / Configuration / manager / proxy .. cleared the field  and applied
error log still shows "Service Unavailable"
Then I put the proxy configuration back

I finally got it to install - but I'm unsure of exactly what fixed the situation. Here are my guesses:

1. waiting 15 minutes after making a change wasn't long enough
2. running the elastic agent setup from a localmachine adminstrator account didn't have enough privilege
3. the firewall field for elastic_agent endpoint might not handle an entry without a newline. I changed it from just 172.16.0.0/15 to:
```
192.168.0.0/16
172.16.0.0/15
10.0.0.0/8
```
I'd like to understand what happened, but ultimately I'm happy that it's working and we can go on with deployment.
(now I've got to go read that discussion I saw about deploying with group policy...)

Thanks for your help.
